apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications.js

```diff
@@ -112,42 +146,89 @@ const sendSubmitMeterReadingsPushNotifications = async () => {
             offset: state.offset,
             pageSize: state.pageSize,
         })
-        const meterReadings = await readMeterReadings({ context, meters })
         const organizations = await readOrganizations({ context, meters })
+        const reportingPeriods = await readMeterReportingPeriods({ context, organizations })
+        const meterReadings = await readMeterReadings({ context, meters, reportingPeriods })
+
+        const metersWithoutReadings = []
+        const periodsByProperty = []
+        const periodsByOrganization = []
+        let defaultPeriod = null
+
+        for (let period of reportingPeriods) {
+            if (!period.organization) defaultPeriod = period
+            else if (!period.property) periodsByOrganization.push(period)
+            else periodsByProperty.push(period)
+        }
+
+        for (const meter of meters) {
+
+
+
+            if (meter.deletedAt) console.log('deletedMeter123', meter)
+
+
 
-        // right join data to metersPage
-        const metersWithoutReadings = rightJoin(
-            meters,
-            meterReadings,
-            (meter, item) => meter.id === item.meter.id,
-            (meter, readings) => ({ meter, readings })
-        )
-            .filter(({ readings }) => readings.length === 0)
-            .map(({ meter }) => meter)
+
+            const period = periodsByProperty.find(({ property }) => property.id === meter.property.id) ??
```

Comment:
what if property field for periodsByProperty item is null? This code gonna fail?

Reply:
But it's impossible. 
`if (!period.property) periodsByOrganization.push(period)`
 this is how this array is filled, falsy value cannot get there

Reply:
if a property null is stored in the MeterReportingPeriod instance, then such a period will fall into periodsByOrganization
because here `!period.property` => `periodsByOrganization`
```
for (let period of reportingPeriods) {
            if (!period. organization) defaultPeriod = period
            else if (!period.property) periodsByOrganization.push(period)
            else periodsByProperty.push(period)
}
```